nevergrad/functions/control/core.py

```diff
@@ -35,25 +35,58 @@ class BaseFunction(ExperimentFunction):
+    deterministic_sim: bool (default: True)
+        Deterministic when initial states are the same for all evaluations
+    noise_level: float (default: 0.)
+        Level (standard deviation) of noise applied to final action
+    intermediate_layer_dim: tuple or None
+        Shape of intermediate layers if exists else None
+    layer_rescaling_coef: tuple or None
+        Scaling coefficient of output layers
+    states_normalization: bool (default: True)
+        Normalize states with `state_mean` and `state_std`
     random_state: int or None
         random state for reproducibility in Gym environment.
     """
 
-    def __init__(self, num_rollouts: int, random_state: tp.Optional[int] = None) -> None:
-        super().__init__(self._simulate, p.Array(shape=self.policy_dim))
+    def __init__(self, num_rollouts: int, activation: str = "tanh", intermediate_layer_dim: tp.Optional[tuple] = None,
+                 deterministic_sim: bool = True, noise_level: float = 0., states_normalization: bool = True,
+                 layer_rescaling_coef: tp.Optional[tuple] = None, random_state: tp.Optional[int] = None) -> None:
+        if intermediate_layer_dim is not None:
+            self.policy_dim = (self.policy_dim[0],) + intermediate_layer_dim + (self.policy_dim[1],) # type: ignore
+        list_parametrizations = [p.Array(shape=(a, b)).set_name(r"layer_{a}_{b}") for a, b in
+                                 zip(self.policy_dim[:-1], self.policy_dim[1:])]
+        parametrization = p.Instrumentation(*list_parametrizations).set_name(self.env_name)
+        super().__init__(self._simulate, parametrization)
         self.num_rollouts = num_rollouts
         self.random_state = random_state
+        self.activation = activation
+        self.states_normalization = states_normalization
+        self.noise_level = noise_level
+        self.deterministic_sim = deterministic_sim
+        self.layer_rescaling_coef = layer_rescaling_coef
+        if layer_rescaling_coef is None: self.layer_rescaling_coef = np.ones(len(self.policy_dim) - 1) # type: ignore
+        self.add_descriptors(num_rollouts=num_rollouts, intermediate_layer_dim=intermediate_layer_dim,
+                             activation=activation, states_normalization=states_normalization,
+                             noise_level=self.noise_level, deterministic_sim=deterministic_sim)
+        if self.noise_level > 0. or not deterministic_sim: self.parametrization.descriptors.deterministic_function = False
         self._descriptors.pop("random_state", None)  # remove it from automatically added descriptors
 
-    def _simulate(self, x: np.ndarray) -> float:
-        env = GenericMujocoEnv(
-            env_name=self.env_name,
-            state_mean=self.state_mean,
-            state_std=self.state_std,
-            num_rollouts=self.num_rollouts,
-            random_state=self.random_state,
-        )
-        return env(x)
+    def _simulate(self, x: tp.Tuple) -> float:
```

Comment:
There is something weird here
Given your instrumentation, you should have
```suggestion
    def _simulate(self, *x: np.ndarray) -> float:
```

Reply:
What I have understood is that, for p.Instrumentation, parameter `x` is a tuple of (args, kwargs) ? thus, x[0] is a list of np.ndarray. I changed parametrization from p.Intrumentation to p.Tuple to be less confusing.